javascript/code-challenges/challenges-01.test.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function named `addOne` that takes an array of numbers, and returns a new array of the numbers, incremented by 1.
+
+Use `forEach` to loop over the input array and work with each value.  Push the new value into a local array. Return the local array;
+------------------------------------------------------------------------------------------------ */
+
+const addOne = (arr) => {
+  const newArray = [];
+  // Solution code here...
+
+  arr.forEach(num => {
+    newArray.push(num + 1);
+  });
+
+  return newArray;
+
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named `addExclamation` that takes an array of strings, and returns a new array of the same strings with an "!" added to the end.
+
+Use `forEach` to loop over the input array. Modify each string, and add the updated value into a local array. Return the local array;
+------------------------------------------------------------------------------------------------ */
+
+const addExclamation = (arr) => {
+  const newArray = []
+  // Solution code here...
+
+  arr.forEach(greet => {
+    newArray.push(`${greet}!`);
+  })
+
+  return newArray;
+
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named `allUpperCase` that takes an array of strings, and returns a new array of the strings converted to upper case.
+
+Use `forEach` to loop over the input array. The modified strings should each be added into a local array. Return that local array.
+------------------------------------------------------------------------------------------------ */
+
+const allUpperCase = (arr) => {
+  const newArray = []
+  // Solution code here...
+
+  arr.forEach(caps => {
```

Comment:
Nice work so far here! 

I wanted to provide you some food for thought regarding variable names. No need to change the code but for this challenge, what are some other names we could have given `caps`? What type of value is `caps`